app/end_block.go

```diff
@@ -0,0 +1,14 @@
+package app
+
+import (
+	appversion "github.com/celestiaorg/celestia-app/x/version"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// EndBlock wraps the BaseApp's end block method. This is done to modify the app
+// version if necessary.
+func (app *App) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBlock) {
+	res = app.BaseApp.EndBlock(req)
+	ctx := app.GetContextForDeliverTx([]byte{1})
+	return appversion.EndBlocker(ctx, app.VersionKeeper, res)
+}
```

Comment:
we have to wrap end block because the base app doesn't actually have a way to modify the app version. It instead relies on the app version being set in `BaseApp`, and then returned during the `Info` call on boot. To avoid that, we modify the consensus param updates and pass them here.

Reply:
I still don't quite get this. Can't you modify the app version in `ResponseEndBlock`?

Reply:
Does it get overwritten somewhere later on?

Reply:
only the events are collected in [EndBlock](https://github.com/celestiaorg/cosmos-sdk/blob/50f4c49034ee97424a43b32e77715f9c8413bb5d/baseapp/abci.go#L205-L208), and [GetConsensusParams](https://github.com/celestiaorg/cosmos-sdk/blob/50f4c49034ee97424a43b32e77715f9c8413bb5d/baseapp/baseapp.go#L438-L469) never checks for version changes. 

changing the app version using `BaseApp` is entirely reliant upon setting the value via [`SetProtocolVersion`](https://github.com/celestiaorg/cosmos-sdk/blob/50f4c49034ee97424a43b32e77715f9c8413bb5d/baseapp/options.go#L113-L116), which isn't passed back to tendermint until `Info`, despite tendermint having the capability to update the version during `EndBlock`

Reply:
so yeah its overwritten. nothing happens because only the result of `GetConsensusParams` is respected, which doesn't check for app vesion

Reply:
Can't we pass the `SetProtocolVersion` function to the version module and just call it in the `EndBlock` :) - Just an idea (It's what the upgrade module does).

BTW, the SDK should really modify the response to hide the fields you can't change to avoid confusion

Reply:
Ah I think I get what you mean. We have to set the app version in two different areas